src/main/java/com/microsoft/aad/msal4j/MSALRuntimeAuthResult.java

```diff
@@ -0,0 +1,112 @@
+package com.microsoft.aad.msal4j;//TODO: will be move to JavaMSALRuntime package, must not reference MSAL Java directly
+
+import com.sun.jna.Structure;
+import com.sun.jna.WString;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
+
+import java.util.concurrent.CompletableFuture;
+
+//FieldOrder needs to explicitly state the public fields that should be in the C structure JNA creates
+@Structure.FieldOrder({"authParamsHandle", "accountHandle", "authResultHandle", "idToken", "accessToken", "accessTokenExpirationTime", "accountId", "accountClientInfo"})
+public class MSALRuntimeAuthResult extends Structure implements MSALRuntimeLibrary.AuthResultCallbackData {
+    public LongByReference authParamsHandle;
+    public LongByReference accountHandle;
+    public LongByReference authResultHandle;
+    public MSALRuntimeUtils.StringByReference idToken = new MSALRuntimeUtils.StringByReference(new WString(""));
+    public MSALRuntimeUtils.StringByReference accessToken = new MSALRuntimeUtils.StringByReference(new WString(""));
+    public LongByReference accessTokenExpirationTime;
+    public MSALRuntimeUtils.StringByReference accountId = new MSALRuntimeUtils.StringByReference(new WString(""));//TODO: split AuthenticationResult into SignInResult (id token/account) and TokenResult (access tokens)? Or keep it simple all-in-one like MSAL Java's AuthenticationResult?
+    public MSALRuntimeUtils.StringByReference accountClientInfo = new MSALRuntimeUtils.StringByReference(new WString(""));
+    private MSALRuntimeLibrary MSALRuntime_INSTANCE;
+    private MSALRuntimeError MSALRuntimeErrorHandler;
+
+    public void setMSALRuntimeLibraryInstance (MSALRuntimeLibrary MSALRuntime_INSTANCE) {
```

Comment:
Is it convention in this library to upper case MSAL? In the Azure SDKs we would likely call this method `setMsalRuntimeLibraryInstance`, so we would only upper-case the M in MSAL.

Reply:
Also, the parameter name here (and below) is weird - can we just call it `instance`?

Reply:
Agreed, and in the latest commit I refactored these sorts of variables to be things like 'setMsalRuntime...' and 'msalRuntimeLibraryInstance'. When initially naming these variables I just followed how I'd normally write MSAL, and was following some conventions I saw in the JNA documentation and MSALRuntime code whenever I did something that referenced something on the C++ side of things.